docs/tutorials/dev_add_test_group.md

```diff
@@ -0,0 +1,2218 @@
+with a file path but rather with a package like in these examples.
+
+The model config options will start with the default set for the E3SM
+component, provided along with the model executable at the end of compilation.
+For MPAS-Ocean and MALI, this will be in 
+`default_inputs/namelist.<component>.forward`.  (For Omega, this has yet to
+be determined.)  Each time a yaml or namelist file is added to a step, the
+model config options changed in that file will override the previous values.
+So the order of the files may matter if the same model config options are
+changed in multiple files in polaris.
+
+Streams are handled a little differently.  Again, the starting point is a set
+of defaults from the E3SM components, such as
+`default_inputs/streams.<component>.forward`.  But in this case, streams are
+only included in the step if they are referenced in one of the yaml or streams
+files added to it.  If you want the default definition of a stream, referring
+to it is enough:
+```yaml
+omega:
+  streams:
+    restart: {}
+```
+
+If you want to change one of its attributes but not its contents, you can do
+that:
+```yaml
+omega:
+  streams:
+    input:
+      filename_template: initial_state.nc
+```
+Other attributes will remain as they are in the defaults.  You can
+change the contents (the variables or arrays) of a stream in addition to the
+attributes.  In this case, the contents you provide will replace the default 
+contents:
+```yaml
+omega:
+  streams:
+    output:
+      type: output
+      filename_template: output.nc
+      output_interval: 0000_00:00:01
+      clobber_mode: truncate
+      contents:
+      - tracers
+      - xtime
+      - normalVelocity
+      - layerThickness
+```
+Finally, you can add completely new streams that don't exist in the default
+model config files to a step by defining all of the relevant streams attributes
+and contents.  We don't demonstrate that in this tutorial.
+
+Another way to set model config options is to use a python dictionary and to 
+call {py:meth}`polaris.ModelStep.add_model_config_options()`.  This is the way 
+to handle namelist options that depend on parameters (such as resolution) that
+are not known in advance.  In this case, we use this technique to set the 
+model config option for the viscosity `config_mom_del2` using the parameter 
+`nu` passed into the constructor (if it is not `None`, indicating that it was 
+not set).
+
+```python
+from polaris.ocean.model import OceanModelStep
+
+
+class Forward(OceanModelStep):
+    def __init__(self, test_case, resolution, name='forward', subdir=None,
+                 ntasks=None, min_tasks=None, openmp_threads=1, nu=None,
+                 run_time_steps=None):
+        ...
+
+        if nu is not None:
+            # update the viscosity to the requested value
+            self.add_model_config_options(options=dict(config_mom_del2=nu))
+```
+
+### Computing the cell count
```

Comment:
I'd move this section toward the end, as an "enhancement". My thinking is that folks would make the simplest possible test work, then if there is time, add in these features.